test/YieldDirectorV2.test.js

```diff
@@ -0,0 +1,817 @@
+    });
+
+    it("should display total donated to recipient", async () => {
+        const principal = `1${e18}`;
+        await tyche.deposit(principal, bob.address);
+        await triggerRebase();
+
+        await expect(await tyche.donatedTo(deployer.address, bob.address)).is.equal(
+            (await gOhm.balanceTo("10000000")).add("1")
+        );
+        await expect(await tyche.totalDonated(deployer.address)).is.equal(
+            (await gOhm.balanceTo("10000000")).add("1")
+        );
+    });
+
+    it("should display total deposited to all recipients", async () => {
+        const principal = `1${e18}`;
+
+        await tyche.deposit(principal, bob.address);
+        await tyche.deposit(principal, alice.address);
+        await triggerRebase();
+
+        await expect(await tyche.depositsTo(deployer.address, bob.address)).is.equal(
+            await gOhm.balanceTo(`10${e9}`)
+        );
+        await expect(await tyche.depositsTo(deployer.address, alice.address)).is.equal(
+            await gOhm.balanceTo(`10${e9}`)
+        );
+
+        await expect(await tyche.depositsTo(bob.address, alice.address)).is.equal("0");
+
+        await expect(await tyche.totalDeposits(deployer.address)).is.equal(
+            await gOhm.balanceTo(`20${e9}`)
+        );
+    });
+
+    it("should display donated amounts across multiple recipients", async () => {
+        const principal = `1${e18}`;
+        await tyche.deposit(principal, bob.address);
+        await tyche.deposit(principal, alice.address);
+
+        await triggerRebase();
+
+        const totalDonation = await gOhm.balanceTo("20000000");
+        await expect(await tyche.totalDonated(deployer.address)).is.equal(totalDonation.add("1"));
+
+        const donationInfo = await tyche.depositInfo("0");
+        const withdrawableBalance = await gOhm.balanceTo(donationInfo.principalAmount);
+        await tyche.withdrawPrincipal("0", withdrawableBalance);
+        await tyche.withdrawPrincipal("1", withdrawableBalance);
+        await expect(await tyche.totalDonated(deployer.address)).is.equal(totalDonation.add("2"));
+
+        await triggerRebase();
+        await expect(await tyche.totalDonated(deployer.address)).is.equal(totalDonation.add("2"));
+
+        await expect(await tyche.totalDonated(deployer.address)).is.equal(totalDonation.add("2"));
+    });
+
+    it("should get all deposited positions", async () => {
+        const principal = `1${e18}`;
+        await tyche.deposit(principal, bob.address);
+        await tyche.deposit(principal, alice.address);
+
+        await triggerRebase();
+
+        const allDeposits = await tyche.getAllDeposits(deployer.address);
+        await expect(allDeposits[0].length).is.equal(2);
+        await expect(allDeposits[0][0]).is.equal(bob.address);
+        await expect(allDeposits[1][0]).is.equal(await gOhm.balanceTo(`10${e9}`));
+    });
+
+    it("test deposit gas", async () => {
+        await tyche.deposit(`1${e18}`, bob.address);
```

Comment:
should these be left in the pr after done testing? it kinda makes the test suite run longer because the beforeAll is quite long.